corehq/apps/sso/backends.py

```diff
@@ -76,8 +77,26 @@ def authenticate(self, request, username, idp_slug, is_handshake_successful):
             is_new_user = False
             web_user = WebUser.get_by_username(username)
         except User.DoesNotExist:
-            user, web_user = self._create_new_user(request, username, async_signup)
-            is_new_user = True
+            try:
+                # the atomic block is necessary to avoid any race conditions
+                # that came up during testing of this feature.
```

Comment:
I would prefer less confidence here. When I come upon a comment like this, and it doesn't match my assumptions about how the code works, I defer to the comment. My concern here is that this hasn't been proven as a solution, and could lead a future developer in the wrong direction if this theory is incorrect.